src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -617,6 +617,21 @@ private void VerifyAndUpdateWrite(int count)
             _responseBytesWritten += count;
         }
 
+        protected void VerifyResponseContentLength()
+        {
+            var responseHeaders = FrameResponseHeaders;
+
+            if (!HttpMethods.IsHead(Method) &&
+                !responseHeaders.HasTransferEncoding &&
```

Comment:
Do you have to check `!responseHeaders.HasTransferEncoding` since your are checking `responseHeaders.HasContentLength`?